### ISSUE TITLE
Add the ability to remove nodes.

### ISSUE THREAD

**Issue by Aaronepower** [utkarshkukreti/select.rs#41]
In [BeautifulSoup](https://www.crummy.com/software/BeautifulSoup/bs4/doc/#decompose) there is the ability to remove nodes from the scraper, this is valuable for removing certain kinds of text or elements from text.

**Comment by utkarshkukreti** [comment]
Yes, I would definitely like to have this feature. Unfortunately, this would require major changes in the internals of the crate and I'm not sure what a good design would look like at this point.

Right now `Document` has a vector of `node::Row` and `node::Node` has a reference to the `Document` and a `usize` index. This means allowing removing/inserting nodes will require some kind of Arena like structure so that removed spots are available for reuse by nodes inserted later. We'll also have to not store a reference to `Document` in `Node` so that the `Document` can be mutated while one of its Node exists. We could have `Node` be just an index like the `petgraph` crate does but that'll make many current APIs verbose, e.g. `document[node].text()` instead of `node.text()`. Or we could just go and wrap everything in `Rc<RefCell<>>` but I'd like to not do that if at all possible.

I'm open to suggestions!

**Comment by sbeckeriv** [comment]
would it be hard to just blank out the contents of the node? Or have a node::RemovedNode?
